File: extensions/shared/src/main/java/app/revanced/extension/youtube/patches/misc/OpenLinksExternallyPatch.java
package app.revanced.extension.youtube.patches.misc;

import app.revanced.extension.youtube.settings.Settings;

@SuppressWarnings("unused")
public class OpenLinksExternallyPatch {

    // renamed from 'enableExternalBrowser'
    public static String openLinksExternally(final String original) {
        if (!Settings.OPEN_LINKS_EXTERNALLY.get())
            return original;

        return "";
    }
}

File: src/main/kotlin/app/revanced/patches/youtube/misc/externalbrowser/OpenLinksExternallyPatch.kt
package app.revanced.patches.youtube.misc.externalbrowser

import app.revanced.patcher.data.BytecodeContext
import app.revanced.patches.youtube.utils.compatibility.Constants.COMPATIBLE_PACKAGE
import app.revanced.patches.youtube.utils.settings.SettingsPatch
import app.revanced.util.patch.BaseBytecodePatch

@Suppress("unused")
object OpenLinksExternallyPatch : BaseBytecodePatch(
    name = "Enable external browser",
    description = "Adds an option to always open links in your browser instead of in the in-app-browser.",
    dependencies = setOf(
        OpenLinksExternallyBytecodePatch::class,
        SettingsPatch::class
    ),
    compatiblePackages = COMPATIBLE_PACKAGE
) {
    override fun execute(context: BytecodeContext) {

        /**
         * Add settings
         */
        SettingsPatch.addPreference(
            arrayOf(
                "SETTINGS: ENABLE_EXTERNAL_BROWSER"
            )
        )

        SettingsPatch.updatePatchStatus(this)
    }
}